internal/consumer/ackMgr.go

```diff
@@ -110,20 +110,27 @@ func (mgr *ackManager) GetAckID(msgSeq int64) (ackID, error) {
 }
 
 // Ack marks the given msgSeqNum as processed
-func (mgr *ackManager) Ack(id ackID) {
-	err := mgr.unackedSeqList.Remove(id.listAddr, id.msgSeq)
+func (mgr *ackManager) Ack(id ackID) (err error) {
+	err = mgr.unackedSeqList.Remove(id.listAddr, id.msgSeq)
 	if err != nil {
-		mgr.tally.Counter(metrics.KafkaPartitionAckErrors)
+		mgr.tally.Counter(metrics.KafkaPartitionAckErrors).Inc(1)
 		mgr.logger.Error("ack error: list remove failed", zap.Error(err))
 	}
+	mgr.tally.Counter(metrics.KafkaPartitionAck).Inc(1)
+	return
 }
 
 // Nack marks the given msgSeqNum as processed, the expectation
 // is for the caller to move the message to an error queue
 // before calling this
-func (mgr *ackManager) Nack(id ackID) {
-	mgr.Ack(id)
-	mgr.tally.Counter(metrics.KafkaPartitionNacks)
+func (mgr *ackManager) Nack(id ackID) (err error) {
+	err = mgr.unackedSeqList.Remove(id.listAddr, id.msgSeq)
```

Comment:
same comment as above. Document what the caller is supposed to do. In the case of a duplicate ack or nack, we don't want the caller to sit in a tight loop and retry - it won't help with anything